src/Cli/dotnet/SlnFileFactory.cs

```diff
@@ -19,87 +54,77 @@ public static string[] ListSolutionFilesInDirectory(string directory, bool inclu
+                jsonDocument = JsonDocument.Parse(File.ReadAllText(filteredSolutionPath));
+                jsonElement = jsonDocument.RootElement;
+                filteredSolutionJsonElement = jsonElement.GetProperty("solution");
+                originalSolutionPath = filteredSolutionJsonElement.GetProperty("path").GetString();
+                originalSolutionPathAbsolute = Path.GetFullPath(originalSolutionPath, Path.GetDirectoryName(filteredSolutionPath));
+                if (!File.Exists(originalSolutionPathAbsolute))
                 {
-                    throw new GracefulException(
-                        CommonLocalizableStrings.CouldNotFindSolutionOrDirectory,
-                        solutionDirectory);
+                    throw new Exception();
                 }
+                filteredSolutionProjectPaths = filteredSolutionJsonElement.GetProperty("projects")
+                    .EnumerateArray()
+                    .Select(project => project.GetString())
+                    .ToArray();
             }
-            catch (ArgumentException)
-            {
+            catch (Exception ex) {
                 throw new GracefulException(
-                    CommonLocalizableStrings.CouldNotFindSolutionOrDirectory,
-                    solutionDirectory);
+                    CommonLocalizableStrings.InvalidSolutionFormatString,
+                    filteredSolutionPath, ex.Message);
             }
 
-            FileInfo[] files = [
-                ..dir.GetFiles("*.sln"),
-                ..(includeSolutionXmlFiles ? dir.GetFiles(".slnx") : [])
-             ];
+            SolutionModel filteredSolution = new SolutionModel();
+            SolutionModel originalSolution = CreateFromFileOrDirectory(originalSolutionPathAbsolute);
 
-            if (files.Length == 0)
+            foreach (var platform in originalSolution.Platforms)
             {
-                throw new GracefulException(
-                    CommonLocalizableStrings.CouldNotFindSolutionIn,
-                    solutionDirectory);
+                filteredSolution.AddPlatform(platform);
             }
-
-            if (files.Length > 1)
+            foreach (var buildType in originalSolution.BuildTypes)
             {
-                throw new GracefulException(
-                    CommonLocalizableStrings.MoreThanOneSolutionInDirectory,
-                    solutionDirectory);
+                filteredSolution.AddBuildType(buildType);
             }
 
-            FileInfo solutionFile = files.Single();
-            if (!solutionFile.Exists)
+            foreach (string path in filteredSolutionProjectPaths)
             {
-                throw new GracefulException(
-                    CommonLocalizableStrings.CouldNotFindSolutionIn,
-                    solutionDirectory);
+                // Normalize path to use correct directory separator
+                string normalizedPath = path.Replace('\\', Path.DirectorySeparatorChar);
```

Comment:
Is this a requirement for `FindProject`? I feel like there might be some utility method for this, but I could be wrong.

Reply:
Yes, tests are failing on *nix, mostly because slnf files usually expect backlashes